File: src/main/java/de/nordgedanken/auto_hotkey/actions/AHKCreateFileAction.java
package de.nordgedanken.auto_hotkey.actions;

import com.intellij.ide.actions.CreateFileFromTemplateAction;
import com.intellij.ide.actions.CreateFileFromTemplateDialog;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.psi.PsiDirectory;
import de.nordgedanken.auto_hotkey.AHKIcons;
import org.jetbrains.annotations.NotNull;

public class AHKCreateFileAction extends CreateFileFromTemplateAction implements DumbAware {
    private final static String CAPTION = "AutoHotKey File";

    public AHKCreateFileAction() {
        super(CAPTION, "", AHKIcons.FILE);
    }

    @Override
    protected void buildDialog(Project project, PsiDirectory directory, CreateFileFromTemplateDialog.Builder builder) {
        builder.setTitle(CAPTION).addKind("Empty File", AHKIcons.FILE, "AutoHotKey File");
    }

    @Override
    protected String getActionName(PsiDirectory directory, @NotNull String newName, String templateName) {
        return CAPTION;
    }

    @Override
    protected boolean isAvailable(DataContext dataContext) {
        return super.isAvailable(dataContext);
    }
}

File: src/main/kotlin/de/nordgedanken/auto_hotkey/ide/actions/AhkCreateFileAction.kt
package de.nordgedanken.auto_hotkey.ide.actions

import com.intellij.ide.actions.CreateFileFromTemplateAction
import com.intellij.ide.actions.CreateFileFromTemplateDialog
import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiDirectory
import de.nordgedanken.auto_hotkey.util.AhkIcons

/**
 * Defines the settings for creating a new Ahk file when you right-click on a file/folder in the project tree
 */
class AhkCreateFileAction : CreateFileFromTemplateAction(CAPTION, "", AhkIcons.FILE), DumbAware {
    override fun buildDialog(project: Project, directory: PsiDirectory, builder: CreateFileFromTemplateDialog.Builder) {
        builder.setTitle(CAPTION).addKind("Empty File", AhkIcons.FILE, CAPTION)
    }

    override fun getActionName(directory: PsiDirectory, newName: String, templateName: String) = CAPTION

    companion object {
        private const val CAPTION = "AutoHotkey File"
    }
}
